### PR TITLE
[SYCL][Fusion] Do not use SPIRV's BuiltInGlobalLinearIdv in remapping

### PR DESCRIPTION
**Pull request by victor-eds**
Using this builtin leads to using SPIR-V's subscript ordering instead of SYCL's, which may lead to performance loss.

As a result, we cannot fuse if ID remapping is used and non-zero offsets are used.

### CODE REVIEW

File: sycl-fusion/test/kernel-fusion/check-remapping.ll [hunk 1]
```diff
@@ -91,6 +86,19 @@ declare !sycl.kernel.fused !13 !sycl.kernel.nd-ranges !15 !sycl.kernel.nd-range
...
+; CHECK-NEXT:    entry:
+; CHECK-NEXT:     %0 = call spir_func i64 @_Z33__spirv_BuiltInGlobalInvocationIdi(i32 2)
+; CHECK-NEXT:     %1 = mul i64 %0, 1
+; CHECK-NEXT:     %2 = call spir_func i64 @_Z33__spirv_BuiltInGlobalInvocationIdi(i32 1)
+; CHECK-NEXT:     %3 = mul i64 %2, 1
+; CHECK-NEXT:     %4 = add i64 %3, %1
+; CHECK-NEXT:     %5 = call spir_func i64 @_Z33__spirv_BuiltInGlobalInvocationIdi(i32 0)
+; CHECK-NEXT:     %6 = add i64 %5, %4
+; CHECK-NEXT:     ret i64 %6
+; CHECK-NEXT:   }
```
Comment (victor-eds) [inline sycl-fusion/test/kernel-fusion/check-remapping.ll:100]: Due to a bug when fusing more than one kernel in a module (not found when coming from the runtime), we are only using one fused kernel. Will fix in a different PR and update test adding the rest of the functions that should be generated.

**Review by sommerlukas — APPROVED** [review]

**Review by Naghasan — APPROVED** [review]

**Review by sommerlukas — APPROVED** [review]

**Review by Naghasan — APPROVED** [review]

**Comment by victor-eds** [comment]
Failing `SYCL :: Basic/gpu_max_wgs_error.cpp` test unrelated to these changes. These were showing up in previous [PRs](https://github.com/intel/llvm/actions/runs/4299730306).

**Comment by dm-vodopyanov** [comment]
> > Failing `SYCL :: Basic/gpu_max_wgs_error.cpp` test unrelated to these changes. These were showing up in previous [PRs](https://github.com/intel/llvm/actions/runs/4299730306).
> 
> @intel/llvm-gatekeepers can I get this merged?

Hi @victor-eds, the merging is blocked by merge conflict,  can you please resolve it?

**Comment by victor-eds** [comment]
> Hi @victor-eds, the merging is blocked by merge conflict, can you please resolve it?

Thanks! Conflict resolved. I'll ping when pipeline passes for merging.

**Comment by victor-eds** [comment]
> Test failure in post-commit: https://github.com/intel/llvm/actions/runs/4386521638/jobs/7680686283
> 
> ```
> Failed Tests (1):
>   SYCL-FUSION :: kernel-fusion/check-remapping.ll
> ```

Thanks for the comment. This should be fixed [here](#8626)

**Pull request merged**